src/hw/hw_mngr.cpp

```diff
@@ -357,3 +463,159 @@ void HWManager::setFanSpeedDelayed()
         setFanSpeed();
     });
 }
+
+
+void HWManager::detectFunsDelayed()
+{
+    static boost::asio::deadline_timer tmr(io);
```

Comment:
I suppose it would be more useful as a part of `Fan` class.
It would lead to several independent timers allow to process each fan in almost parallel.

Reply:
More separate timers, more separate states, more different flow cases (and get signal race). And HWManager must control it on boot time. Do we really need to increase algorithm complexity?   
At current fix we alreaady process all fans in parallel.